app/jobs/scheduled_publishing_job.rb

```diff
@@ -9,15 +9,19 @@ class ScheduledPublishingJob < ApplicationJob
   discard_and_log(ActiveRecord::RecordNotFound)
 
   def perform(id)
-    Edition.find_and_lock_current(id: id) do |edition|
-      next if no_longer_schedulable?(edition)
+    edition = Edition.find_current(id: id)
+    user = edition.status.created_by
 
-      user = edition.status.created_by
-      reviewed = edition.status.details.reviewed
+    Edition.transaction do
+      edition.lock!
```

Comment:
I'd suggest we set edition here with `edition = edition.find_and_lock_current(id: id)` this'll still be in scope after the transaction block.

Reply:
Done, sort of! I've gone with the normal approach `Edition.find_and_lock_current(id: ..) { ... }` and just assigned the return to a variable, we can't actually call `find_and_lock_current` without passing in a block.